components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt

```diff
@@ -189,12 +193,17 @@ class AppLinksUseCases(
      *
      * This does not do any additional UI other than the chooser that Android may provide the user.
      */
-    class OpenAppLinkRedirect internal constructor(
+    inner class OpenAppLinkRedirect internal constructor(
         private val context: Context
     ) {
         operator fun invoke(appIntent: Intent?, failedToLaunchAction: () -> Unit = {}) {
             appIntent?.let {
                 try {
+                    val scheme = appIntent.data?.scheme
+                    if (scheme != null && alwaysDeniedSchemes.contains(scheme)) {
+                        return
+                    }
```

Comment:
I do not have the full context here - but is that not too late? If we already know that we are not going to open this - should we ever allow anyone to get here?

Reply:
technically it can still be called directly by the caller.  Fenix currently calls directly like this `appLinksUseCases.openAppLink.invoke(redirect.appIntent)`. Which can still cause a crash when the redirect.appIntent is a file.